app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -146,14 +146,15 @@ def fetch_server_entities
       end
 
       def fetch_availability
-        metric_ids = {}
+        resources_by_metric_id = {}
         @data[:middleware_deployments].each do |deployment|
           metric_id = @ems.build_metric_id('A', deployment, 'Deployment Status~Deployment Status')
-          metric_ids[metric_id] = deployment
+          resources_by_metric_id[metric_id] = deployment
         end
-        availabilities = @ems.metrics_client.avail.raw_data(metric_ids.keys, :limit => 1, :order => 'DESC')
-        availabilities.each do |availability|
-          metric_ids[availability['id']][:status] = process_availability(availability['data'].first)
+        unless resources_by_metric_id.empty?
+          availabilities = @ems.metrics_client.avail.raw_data(resources_by_metric_id.keys,
+                                                              :limit => 1, :order => 'DESC')
+          parse_availability availabilities, resources_by_metric_id
```

Comment:
Not really, fetch_availability will fetch from provider and parse_availability will parse and put results on resource hash.